lib/inventory/inventory.go

```diff
@@ -70,12 +70,13 @@ type DownstreamHandle interface {
 	CloseContext() context.Context
 	// Close closes the downstream handle.
 	Close() error
-	// SendGoodbye indicates the downstream half of the connection is terminating. This
-	// has no impact on the health of the inventory control stream, nor does it perform
-	// any clean up of the connection. A Goodbye is merely information so that the
-	// upstream half of the connection may take different actions when the downstream
-	// half of the connection is shutting down for good vs. restarting.
-	SendGoodbye(context.Context) error
+	// SendGoodbye indicates the downstream half of the connection is starting the
+	// termination process. This has no impact on the health of the inventory control
+	// stream, nor does it perform any clean up of the connection.
+	// In case of soft-reloads, the termination process can take up to 30 hours.
+	// The Goodbye message may indicate the reason for the connection termination
+	// (shutdown versus soft-reload).
+	SendGoodbye(context.Context, bool, bool) error
```

Comment:
Two minor nits:

- It might be better to name this something like `SetGoodbye`.  It's not intuitive that a `Send*` method would store the value and potentially resend it multiple times.
- `bool, bool` is a bit of an unfortunate function signature.  It might be worth just actually accepting a goodbye message here.

Reply:
>It might be better to name this something like SetGoodbye. It's not intuitive that a Send* method would store the value and potentially resend it multiple times.

What about `Goodbye()` or `SetAndSendGoodbye`? I feel like `SetGoodbye` is not clear either about the fact this function synchronously sends the message and blocks until it grabs a sender.

Reply:
+1 for `SetAndSendGoodbye`, with the current semantics.